components/Dialogs/src/private/MDCAlertControllerView+Private.m

```diff
@@ -422,24 +451,54 @@ - (void)layoutSubviews {
   boundsSize.width = boundsSize.width - MDCDialogContentInsets.left - MDCDialogContentInsets.right;
   CGSize titleSize = [self.titleLabel sizeThatFits:boundsSize];
   titleSize.width = boundsSize.width;
+
+  CGSize titleIconSize = CGSizeZero;
+  if (self.titleIconImageView != nil) {
+    // TODO(galiak): Have title-icon size respond to dynamic type or device screen size, once this:
+    // https://github.com/material-components/material-components-ios/issues/5198 is resolved.
+    titleIconSize = self.titleIconImageView.image.size;
+  }
+
   CGSize messageSize = [self.messageLabel sizeThatFits:boundsSize];
   messageSize.width = boundsSize.width;
   boundsSize.width = boundsSize.width + MDCDialogContentInsets.left + MDCDialogContentInsets.right;
 
   CGFloat contentInternalVerticalPadding = 0.0;
-  if (0.0 < titleSize.height && 0.0 < messageSize.height) {
+  if ((0.0 < titleSize.height || 0.0 < titleIconSize.height) && 0.0 < messageSize.height) {
     contentInternalVerticalPadding = MDCDialogContentVerticalPadding;
   }
+  CGFloat contentTitleIconVerticalPadding = 0.0f;
+  if (0.0 < titleSize.height && 0.0 < titleIconSize.height) {
+    contentTitleIconVerticalPadding = MDCDialogTitleIconVerticalPadding;
+  }
 
-  CGRect titleFrame = CGRectMake(MDCDialogContentInsets.left, MDCDialogContentInsets.top,
-                                 titleSize.width, titleSize.height);
+  CGFloat titleTop =
+      MDCDialogContentInsets.top + contentTitleIconVerticalPadding + titleIconSize.height;
+  CGRect titleFrame =
+      CGRectMake(MDCDialogContentInsets.left, titleTop, titleSize.width, titleSize.height);
   CGRect messageFrame = CGRectMake(MDCDialogContentInsets.left,
                                    CGRectGetMaxY(titleFrame) + contentInternalVerticalPadding,
                                    messageSize.width, messageSize.height);
 
   self.titleLabel.frame = titleFrame;
   self.messageLabel.frame = messageFrame;
 
+  if (self.titleIconImageView != nil) {
+    // match the titleIcon alignment to the title alignment
```

Comment:
This seems strange to me, don't we always want to center the icon?

Reply:
There are actually no specifications for this scenario. The requirements only cover 2 modes - (1) a centered icon with a centered title, or (2) a left/natural aligned title with no icon.   All other options are available in MDC because the 2 properties (icon and title-alignment) are independent of each other.  Having the same alignment for the 2 looks better.  See the 2 screenshots in top comment for a demonstration.